src/Database/Relations/DefinedConstraints.php

```diff
@@ -118,7 +118,7 @@ public function addDefinedConstraintsToQuery($query, $args = null)
          * Scope
          */
         if ($scope = array_get($args, 'scope')) {
-            $query->$scope($this->parent);
+            $this->getRelated()->$scope($query);
```

Comment:
I'm not understanding how this would work. This would now call the scope method on a model object instead of a eloquent query builder to be passed to a model object, and when you call `$model->{$scope}` the internal magic will instantiate a new base query builder, and pass that query builder instance to the actual method `$model->scope{$scope}($query);`.

If I'm following correctly, I believe that this line would actually have to be

```php
$scopeMethod = 'scope' . ucfirst($scope);
$this->getRelated()->{$scopeMethod}($query, $this->parent);
```

Additionally, if we're to handle this in an even more laravel-y way, I think the correct way of manually calling a model scope without an eloquent builder instance is `$this->getRelated()->scopes([$scope => [$this->parent]]);` (https://github.com/laravel/framework/blob/6.x/src/Illuminate/Database/Eloquent/Builder.php#L906) (which also does a bunch of other things to improve stability).

Reply:
@bennothommo @mjauvin any thoughts on the above?